File: app/src/main/java/com/berkatfaatulohalawa1711010164/facevoting/helper/ErrorHelper.java
package com.berkatfaatulohalawa1711010164.facevoting.helper;

import com.berkatfaatulohalawa1711010164.facevoting.model.ErrorModel;

import java.io.IOException;
import java.lang.annotation.Annotation;

import okhttp3.ResponseBody;
import retrofit2.Converter;
import retrofit2.Response;

public class ErrorHelper {
    public static ErrorModel parseError(Response<?> response) {
        Converter<ResponseBody, ErrorModel> converter =
                ServiceError.retrofit
                        .responseBodyConverter(ErrorModel.class, new Annotation[0]);
        ErrorModel error ;
        try {
            if(response.errorBody() != null) {
                error = converter.convert(response.errorBody());
            }  else {
                error = null;
            }
        } catch (IOException e) {
            return new ErrorModel("error");
        }
        return error;
    }
}

File: android/facevoting/app/src/main/java/com/berkatfaatulohalawa1711010164/facevoting/helper/ErrorHelper.kt
package com.berkatfaatulohalawa1711010164.facevoting.helper

import com.berkatfaatulohalawa1711010164.facevoting.model.ErrorModel
import retrofit2.Response
import java.io.IOException
import java.lang.annotation.Annotation

object ErrorHelper {
    fun parseError(response: Response<*>): ErrorModel {
        val converter = ServiceError.retrofit
            .responseBodyConverter<ErrorModel>(ErrorModel::class.java, arrayOfNulls<Annotation>(0))
        return try {
            if (response.errorBody() != null) {
                converter.convert(response.errorBody()!!) ?: ErrorModel("error")
            } else {
                ErrorModel("error")
            }
        } catch (e: IOException) {
            ErrorModel("error")
        }
    }
}
